src/common/transformations/src/transformations/faster_hash.cpp

```diff
@@ -0,0 +1,422 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/ops.hpp"
+#include "ngraph/opsets/opset1.hpp"
+#include "openvino/op/util/framework_node.hpp"
+#include "transformations/hash.hpp"
+
+using namespace ngraph;
+
+namespace {  // helpers
+
+template <typename T>
+static uint64_t hash_combine(uint64_t seed, const T& a) {
```

Comment:
Can we use: https://github.com/openvinotoolkit/openvino/blob/master/src/common/util/include/openvino/util/common_util.hpp#L40